ios/RNOneginiSdk.swift

```diff
@@ -176,21 +182,22 @@ class RNOneginiSdk: RCTEventEmitter, ConnectorToRNBridgeProtocol {
             reject(String(WrapperError.malformedUrl.code), WrapperError.malformedUrl.description, WrapperError.malformedUrl)
             return
         }
-
-        if (bridgeConnector.toRegistrationConnector.registrationHandler.processRedirectURL(urlOBject)) {
-            resolve(nil)
-        } else {
-            reject(String(WrapperError.registrationNotInProgress.code), WrapperError.registrationNotInProgress.description, WrapperError.registrationNotInProgress)
+        do {
+            try bridgeConnector.toRegistrationConnector.registrationHandler.processRedirectURL(urlOBject)
+            return resolve(nil)
+        } catch {
+            reject(String(error.code), error.localizedDescription, error)
         }
     }
 
     @objc
     func cancelRegistration(_ resolve: @escaping RCTPromiseResolveBlock,
                             rejecter reject: @escaping RCTPromiseRejectBlock) -> Void {
         let registrationHandler = bridgeConnector.toRegistrationConnector.registrationHandler
-        var canceled = registrationHandler.cancelBrowserRegistration()
-        canceled = registrationHandler.cancelPinCreation() || canceled
-        canceled = registrationHandler.cancelCustomRegistration() || canceled
+        var canceled = false
+        if (try? registrationHandler.cancelBrowserRegistration()) != nil { canceled = true }
+        if (try? registrationHandler.cancelPinCreation()) != nil { canceled = true }
+        if (try? registrationHandler.cancelCustomRegistration()) != nil { canceled = true }
```

Comment:
```
do {
  try? registrationHandler.cancelBrowserRegistration() 
  try? registrationHandler.cancelPinCreation()
  try? registrationHandler.cancelCustomRegistration()
} catch {
  canceled = true
}
```

Reply:
@kfa-onewelcome this code has been reworked here https://github.com/onewelcome/sdk-react-native/pull/96